### PR TITLE
Auto-update frontend on PUT/POST/DELETE

### LINKED ISSUE #111: Auto-update frontend on backend change
Similar to #85, but for the Document dashboard when a Field Partner uploads a file (should go to Pending) and for the Document Class page when a Document Class is added/edited/deleted.

### PR DESCRIPTION
**Pull request by arpanlaha**
Resolves #111
Also implements a loading screen to account for wait time between request and update.

### CODE REVIEW

File: frontend/src/components/Dashboard.js [hunk 1]
```diff
@@ -98,22 +84,34 @@ class Dashboard extends React.Component {
...
-                    <DocumentList documents={this.props.documents[key]} status={key} />
-                  </Col>
-                )
-              })
-            : null}
-        </Row>
-        <NotificationsBar />
-      </Container>
-    )
+    if (this.props.loading) {
```
Comment (arpanlaha) [inline frontend/src/components/Dashboard.js]: Show loading icon if loading prop is true.

File: frontend/src/components/DocumentClass.js [hunk 2]
```diff
@@ -65,27 +72,43 @@ class DocumentClass extends Component {
     this.setState({ description: event.target.value })
   }
 
-  handleSubmit() {
-    updateDocumentClass(
+  async handleSubmit() {
+    this.props.beginLoading()
```
Comment (arpanlaha) [inline frontend/src/components/DocumentClass.js]: set loading true at beginning and false after all requests have completed

File: frontend/src/components/DocumentClass.js [hunk 3]
```diff
@@ -65,27 +72,43 @@ class DocumentClass extends Component {
...
+    }
+    this.props.endLoading()
     this.uploadToggle()
   }
 
   // for if a Document Class is being submitted without an example file
   // may be removed later
-  handleSubmitNoFile() {
-    updateDocumentClass(
+  async handleSubmitNoFile() {
```
Comment (rgychiu) [inline frontend/src/components/DocumentClass.js]: Can this function be consolidated with the handleSubmit function? It seems like the only difference here is that some of the fields is null, which can be handled in the function above.

Comment (arpanlaha) [reply frontend/src/components/DocumentClass.js]: Fixed

**Pull request merged**